### PR TITLE
Spec update for ConvolutionOp: Clarify permute

### PR DESCRIPTION
**Pull request by sdasgup3**
fixes https://github.com/openxla/stablehlo/issues/1338
Please refer to the https://github.com/openxla/stablehlo/issues/1338 for more details.

### CODE REVIEW

File: docs/spec.md [hunk 1]
```diff
@@ -1880,6 +1880,7 @@ This reframing uses the following helper functions:
 
 * `lhs_shape(n, hw, c) = permute([n] + hw + [c], [input_batch_dimension] + input_spatial_dimensions + [input_feature_dimension])`.
 * `result_shape(n1, hw, c1) = permute([n1] + hw + [c1], [output_batch_dimension] + output_spatial_dimensions + [output_feature_dimension])`.
+* `permute([j0, j1, ..., jR-1], permutation)` produces `[i0, i1, ..., iR-1]` such that `j[d] = i[permutation[d]]`.
```
Comment (ghpvnist) [inline docs/spec.md:1883]: `... for all d in [0, size(permutation)).`

**Review by ghpvnist — APPROVED** [review]
Minor comment

File: docs/spec.md (hunk 1 above)
Comment (sdasgup3) [reply docs/spec.md]: Done. Thanks!

File: docs/spec.md [hunk 2]
```diff
@@ -1880,6 +1880,7 @@ This reframing uses the following helper functions:
 
 * `lhs_shape(n, hw, c) = permute([n] + hw + [c], [input_batch_dimension] + input_spatial_dimensions + [input_feature_dimension])`.
 * `result_shape(n1, hw, c1) = permute([n1] + hw + [c1], [output_batch_dimension] + output_spatial_dimensions + [output_feature_dimension])`.
+* `permute([j0, j1, ..., jR-1], permutation)`, with `size(permutation) = R`, produces `[i0, i1, ..., iR-1]` such that `j[d] = i[permutation[d]] for all d in [0, R)`.
```
Comment (burmako) [inline docs/spec.md:1883]: Proposed rewording: 

```
`permute([j0, j1, ..., jR-1], permutation) = [i0, i1, ..., iR-1]` where `j[d] = i[permutation[d]]`.
```

  1. Saying `=` instead of "produces" is consistent with the rest of the spec.
  2. Given the overhead of saying "for all...", I think we should shorten this to the original formulation, which is consistent with the spec of stablehlo.transpose.

**Review by burmako — CHANGES_REQUESTED** [review]

File: docs/spec.md (hunk 2 above)
Comment (sdasgup3) [reply docs/spec.md]: Thanks!
Done.
